exchanges/graphcache/src/operations/query.ts

```diff
@@ -356,6 +356,9 @@ const readSelection = (
   let node: FormattedNode<FieldNode> | void;
   const output = InMemoryData.makeData(input);
   while ((node = iterate()) !== undefined) {
+    const fieldDirectives = Object.keys(node._directives || {}).map(x => x);
+    const storeDirective = fieldDirectives.find(x => store.directives[x]);
```

Comment:
Hm 🤔 This makes me think a bit. I'm not a huge fan of the extra `find` and `keys` work here.

The former should just be replaceable by `getDirectives(node)` which now exists to always return the object.

I'd argue, we should then `for (const name in)` loop over the `fieldDirectives` (as an object) and potentially allow multiple directives to be executed.

This would also require us to cover:
- the field directive not having an entry in `store.directives`
- multiple directives being used

I'd argue multiple directives being used simultaneously should either be a warning or it should be allowed. I'm not sure if there are cases where composition makes sense, but it's possible.

We could just not allow that for now, reuse the same warning code even and add a different message for it.

The minimum here is though I think to just discard the `fieldDirectives` / `storeDirective` split and just find the directive function immediately:

```
function getFieldDirective(node: FormattedNode<FieldNode>, store: Store) {
  for (const name in getDirectives(node)) {
    if (name === 'include' || name === 'skip') continue;
    if (store.directives[name])
      return store.directives[name];
  }
}
```

wdyt?

Reply:
Handling multiple directives is technically allowed according to the spec so I don't mind implementing it

Reply:
took a look at the code, I think I can't find a way to reconcile how to actually chain values, i.e. how to pipe the output of one directive into the next as an input, since that's not how resolvers work, so I reckon, we just drop it for now 🤔 

re. the former, I think there's a slight refactor that can be made to help with that, but also something that can be done after this is merged, since it's probably a good time for me to scan the file for other changes that can be made again 👀

Reply:
@kitten if you want me to refactor other things feel free to point htem out